internal/handlers/tigris/msg_delete.go

```diff
@@ -107,27 +118,61 @@ func (h *Handler) MsgDelete(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
 		}
 
 		if resDocs, err = common.LimitDocuments(resDocs, limit); err != nil {
-			return nil, err
+			return err
 		}
 
+		// if no field is matched in a row, go to the next one
 		if len(resDocs) == 0 {
-			continue
+			return nil
 		}
 
 		res, err := h.delete(ctx, fp, resDocs)
 		if err != nil {
-			return nil, err
+			return err
 		}
 
 		deleted += int32(res)
+
+		return nil
 	}
 
+	delErrors := new(common.WriteErrors)
+
 	var reply wire.OpMsg
+
+	// process every delete filter
+	for i := 0; i < deletes.Len(); i++ {
+		err = processQuery(i)
+		if err != nil {
+			delErrors.Append(err, int32(i))
+
+			// Delete statements in the `deletes` field are not transactional.
+			// It means that we run each delete statement separately.
+			// If `ordered` is set as `true`, we don't execute the remaining statements
+			// after the first failure.
+			// If `ordered` is set as `false`,  we execute all the statements and return
+			// the list of errors corresponding to the failed statements.
+			if ordered {
+				break
+			}
+		}
+	}
+
+	// minimal reply document
+	replyDoc := must.NotFail(types.NewDocument(
+		"n", deleted,
+		"ok", float64(1),
+	))
+
+	// if there are delete errors append writErrors field
+	if len(*delErrors) > 0 {
+		protoErr, _ := common.ProtocolError(delErrors)
+		we := must.NotFail(common.AssertType[*types.Array](must.NotFail(protoErr.Document().Get("writeErrors"))))
```

Comment:
Let's make it the way how we made it for the `pg` handler.

Reply:
nice catch, sorry for overlooking